### PR TITLE
replaced Java download helper with pure python

### PR DESCRIPTION
**Pull request by dougxc**
@ldaynes, this also addresses #51.

### CODE REVIEW

File: mx.py [hunk 1]
```diff
@@ -2096,15 +2096,15 @@ def _get_path_in_cache(name, sha1, urls, ext=None):
...
-    _, binDir = _compile_mx_class('URLConnectionDownload')
-    command = [get_jdk(tag=DEFAULT_JDK_TAG).java, '-cp', _cygpathU2W(binDir), 'URLConnectionDownload']
-    command += ['--no-progress', '--skip-download']
-    command += urls
-    def devnull(s):
-        pass
-    return run(command, err=devnull, nonZeroIsFatal=False) == 0
+    for url in urls:
+        try:
+            urllib2.urlopen(url, timeout=0.5).close()
```
Comment (gilles-duboscq) [inline mx.py]: 0.5s timeout is rather short, do we really need an explicit timeout here?

Comment (dougxc) [inline mx.py]: It was 0.3 before this change. In any case I think an explicit timeout is better than the default which seems to be very long.

Comment (gilles-duboscq) [inline mx.py]: Ok, but shouldn't it be the same timeouts as the others (10)?

Comment (olpaw) [inline mx.py]: I use download_file_exists to probe if the maven artifacts are already existing. So if the probing takes longer than the writing (in the common case) then it kinda defeats the purpose.

Comment (gilles-duboscq) [inline mx.py]: ok

Comment (dougxc) [inline mx.py]: Is that an accept then?

Comment (gilles-duboscq) [inline mx.py]: yes

**Pull request merged**